packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1435,92 +1424,69 @@ mixin WidgetInspectorService {
     return _followDiagnosticableChain(chain.reversed.toList());
   }
 
+  DiagnosticsSerialisationDelegate _configToDelegate(_SerializeConfig config) {
+    final Set<DiagnosticsNode> nodesCreatedByLocalProject = <DiagnosticsNode>{};
+    return DiagnosticsSerialisationDelegate(
+      subtreeDepth: config.subtreeDepth,
+      includeProperties: config.includeProperties,
+      expandPropertyValues: config.expandPropertyValues,
+      additionalNodeProperties: (DiagnosticsNode node, DiagnosticsSerialisationDelegate _) {
+        final Map<String, Object> result = <String, Object>{};
+        final Object value = node.value;
+        if (config.interactive) {
+          result['objectId'] = toId(node, config.groupName);
+          result['valueId'] = toId(value, config.groupName);
+        }
+        if (config.summaryTree) {
+          result['summaryTree'] = true;
+        }
+        final _Location creationLocation = _getCreationLocation(value);
+        if (creationLocation != null) {
+          result['locationId'] = _toLocationId(creationLocation);
+          result['creationLocation'] = creationLocation.toJsonMap();
+          if (_isLocalCreationLocation(creationLocation)) {
+            nodesCreatedByLocalProject.add(node);
+            result['createdByLocalProject'] = true;
+          }
+        }
+        return result;
+      },
+      filterChildren: (List<DiagnosticsNode> children, DiagnosticsNode parent, DiagnosticsSerialisationDelegate _) {
```

Comment:
when I see this many named parameters with Function implementations I really think we should be just defining a class.

Reply:
I don't understand what you mean here. Isn't the DiagnosticsSerialisationDelegate that class?

Reply:
I mean maybe make DiagnosticsSerializationDelegate an abstract class instead of specifying an abstract class the hard way by having a bunch of highly related function callbacks.